src/cpp/link_creation_agent/link_create_template.cc

```diff
@@ -0,0 +1,243 @@
+#include "link_create_template.h"
+
+#include <algorithm>
+#include <iostream>
+#include <stdexcept>
+
+using namespace link_creation_agent;
+
+bool is_number(const std::string& s) {
```

Comment:
Util functions outside classes should be declared `static` to prevent name clashing during linking.

```
static  bool is_number(const std::string& s) {
```

You can use them the same way you use non-static functions. The `static` keyword here means only that the function symbol is not supposed to be exposed outside this `.o` when linking to build an executable.

If you have purposely created a function outside classes which is supposed to be used in other files, you are supposed to prefix the function name with a namespace as you do with class names and put the function signature in the `.h`.